### PR TITLE
oils-for-unix 0.22.0 (new formula)

### PR DESCRIPTION
**Pull request by lishaduck**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This is sort of a duplicate of of [`oil`](https://formulae.brew.sh/formula/oil). Oils for Unix is a project that maintains two shells that run under one interpreter. The original project was called Oil and was written in Python, and it was rebranded as Oils for Unix with the C++ rewrite. [It's recommended](https://www.oilshell.org/release/0.22.0/#Packaging:~:text=The%20oil%20tarball%20is%20the%20slow%20reference%20interpreter%20based%20on%20CPython.) to use the `oil` package for the old Python implementation and `oils-for-unix` for the C++ implementation. If y'all would prefer, I can deprecate or rename `oil`, though that would violate the packaging guidelines, so I haven't.

Additional sources on the rename and C++ installation instructions:
- <https://www.oilshell.org/blog/2023/03/rename.html>
- <https://www.oilshell.org/blog/2023/06/release-0.16.0.html>
- <https://www.oilshell.org/blog/2024/03/release-0.21.0.html>
- <https://www.oilshell.org/release/0.22.0/doc/INSTALL.html>
- <https://github.com/oilshell/oil/blob/master/README-native.txt>

### CODE REVIEW

**Comment by lishaduck** [comment]
The commit style check dislikes that I made `oil` and `oils-for-unix` conflict. Should I extract that, or is it fine?

**Comment by chenrui333** [comment]
> The commit style check dislikes that I made `oil` and `oils-for-unix` conflict. Should I extract that, or is it fine?

that should be done as followup as of now.

File: Formula/o/oil.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ class Oil < Formula
   depends_on "readline"
 
   conflicts_with "omake", because: "both install 'osh' binaries"
+  conflicts_with "oils-for-unix", because: "it's the modern C++ interpreter"
```
Comment (chenrui333) [inline Formula/o/oil.rb:25]: made followup pr to address conflicts
```suggestion
```

File: Formula/o/oils-for-unix.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  license "Apache-2.0"
+
+  livecheck do
+    url "https://www.oilshell.org/releases.html"
+    regex(/href=.*?oils-for-unix[._-]v?(\d+(?:\.\d+)+)\.t/i)
+  end
+
+  depends_on "readline"
+
+  conflicts_with "oil", because: "it's the old CPython-based interpreter"
```
Comment (chenrui333) [inline Formula/o/oils-for-unix.rb:15]:
```suggestion
```

**Review by chenrui333 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**